DESCRIPTION

```diff
@@ -19,14 +19,17 @@ Depends:
     R (>= 4.0)
 Imports:
     checkmate,
+    cli,
```

Comment:
can we remove `cli` dependency? It's a big package

Reply:
if we do then we don't fix this:
https://github.com/insightsengineering/teal.code/issues/86 - unless you want to fix it a different way

Reply:
that's my question - can we do (name that thing) differently not involving cli package?

Reply:
so this is what we call:

```
> cli::ansi_strip
function (string, sgr = TRUE, csi = TRUE, link = TRUE) 
{
    if (!is.character(string)) 
        string <- as.character(string)
    string <- enc2utf8(string)
    stopifnot(is_flag(sgr), is_flag(csi), is_flag(link))
    clean <- .Call(clic_ansi_strip, string, sgr, csi, link)
    class(clean) <- setdiff(class(clean), c("cli_ansi_string", 
        "ansi_string"))
    clean
}
<bytecode: 0x0000018dd338c3e0>
<environment: namespace:cli>
```

Reply:
and dplyr imports cli so in practice ppl using this package will have cli installed but if there's an easy way round this then happy to change (or put issue in bnacklog to change)

Reply:
we could add it into suggests like rlang and only call it if installed?

Reply:
Here I am more concerned about the package size and the installation process. Up to you how would you like to process it.

Reply:
> we could add it into suggests like rlang and only call it if installed?

This only makes sense if a particular piece of code is unlikely to be called assuming _standard_ workflow. E.g. a _wide_ tree-based workflow / case when where there are many many possibilities. Here, if I understand everything correctly it's a part of standard logic so it's very likely that it will be executed.

Reply:
so I assume that these funny terminal printing characters added by rlang are only added if cli is installed (cli is a suggests in rlang only) - therefore I think it makes sense for us to use suggests as well - as striping these terminal printing characters out only makes sense if cli (via rlang) added them in the first place

Reply:
so rlang is only in the suggests but needs to be imported - whoops